marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/EntitySearchController.java

```diff
@@ -138,31 +138,40 @@ public JsonNode getFacetValuesRange(@RequestBody @ApiParam(hidden = true) JsonNo
     @ResponseBody
     @Secured("ROLE_exportEntityInstances")
     @ApiOperation("Returns CSV data")
-    public ResponseEntity<StreamingResponseBody> export(@RequestParam String queryDocument, @RequestParam String fileType, @RequestParam(required = false) Long limit, final HttpServletResponse response) {
-        StreamingResponseBody stream = out -> {
-            newEntitySearchManager().exportByQuery(new ObjectMapper().readTree(queryDocument), fileType, limit, out, response);
-        };
-
+    public ResponseEntity<StreamingResponseBody> export(@RequestParam String queryDocument,
+                                                        @RequestParam String fileType,
+                                                        @RequestParam(required = false) Long limit,
+                                                        final HttpServletResponse response,
+                                                        @RequestParam(defaultValue = "final") String database) {
+        StreamingResponseBody stream = out -> newEntitySearchManager(database).exportByQuery(new ObjectMapper().readTree(queryDocument), fileType, limit, out, response);
         return ResponseEntity.ok(stream);
     }
 
     @RequestMapping(method = RequestMethod.GET, value = "/export/query/{queryId}")
     @ResponseBody
     @Secured("ROLE_exportEntityInstances")
     @ApiOperation("Returns CSV data")
-    public ResponseEntity<StreamingResponseBody> exportSavedQuery(@PathVariable String queryId, @RequestParam String fileType, @RequestParam(required = false) Long limit, final HttpServletResponse response) {
-        StreamingResponseBody stream = out -> {
-            newEntitySearchManager().exportById(queryId, fileType, limit, out, response);
-        };
-
+    public ResponseEntity<StreamingResponseBody> exportSavedQuery(@PathVariable String queryId,
+                                                                  @RequestParam String fileType,
+                                                                  @RequestParam(required = false) Long limit,
+                                                                  final HttpServletResponse response,
+                                                                  @RequestParam(defaultValue = "final") String database) {
+        StreamingResponseBody stream = out -> newEntitySearchManager(database).exportById(queryId, fileType, limit, out, response);
         return ResponseEntity.ok(stream);
     }
 
-    private EntitySearchManager newEntitySearchManager() {
-        return new EntitySearchManager(getHubClient());
+    private EntitySearchManager newEntitySearchManager(String database) {
+        return new EntitySearchManager(getHubClient(), database);
     }
 
     private EntitySearchService getEntitySearchService() {
+        return getEntitySearchService("final");
+    }
+
+    private EntitySearchService getEntitySearchService(String database) {
+        if(database.equalsIgnoreCase("staging")) {
```

Comment:
To ensure an NPE is never possible here, do "staging".equalsIgnoreCase(database)